lib/ProductOpener/Display.pm

```diff
@@ -1069,6 +1094,60 @@ sub display_no_index_page_and_exit () {
 	exit();
 }
 
+=head2 display_robots_txt_and_exit ($request_ref)
+
+Return robots.txt page and exit.
+
+robots.txt is dynamically generated based on lc, it's content depends on $request_ref:
+- if $request_ref->{deny_all_robots_txt} is 1: a robots.txt where we deny all traffic
+  combinations.
+- otherwise: the standard robots.txt. We disallow indexing of most facet pages, the
+  exceptions can be found in ProductOpener::Config::index_tag_types
+
+=cut
+
+sub display_robots_txt_and_exit ($request_ref) {
+	my $template_data_ref = {facets => []};
+	my $vars = {deny_access => $request_ref->{deny_all_robots_txt}, disallow_paths_localized => []};
+
+	foreach my $type (keys %tag_type_singular) {
```

Comment:
Addiing a sort so that we always get the same order in the robots.txt file (useful for testing)

```suggestion
	foreach my $type (sort keys %tag_type_singular) {
```